src/components/inputs/Select.tsx

```diff
@@ -254,22 +273,28 @@ const Select = React.forwardRef<HTMLDivElement, SelectProps>(function SelectFn(
 	useEffect(() => {
 		if (selection) {
 			if (multiple) {
-				dispatchSelected({
-					multiple: true,
-					items: selection as SelectItem[],
-					type: SELECT_ACTION.SET
-				});
-			} else {
+				if (selection instanceof Array) {
+					dispatchSelected({
+						multiple: true,
+						items: selection,
+						type: SELECT_ACTION.SET
+					});
+				}
+			} else if (!(selection instanceof Array)) {
 				dispatchSelected({
 					multiple: false,
-					item: selection as SelectItem
+					item: selection
 				});
 			}
 		}
 	}, [multiple, selection]);
 
 	useEffect(() => {
-		onChange?.(multiple ? selected : selected[0]?.value ?? null);
+		if (multiple) {
+			(onChange as MultipleSelectionOnChange)?.(selected);
+		} else {
+			(onChange as SingleSelectionOnChange)?.(selected[0]?.value ?? null);
+		}
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [selected, multiple]);
```

Comment:
I think we should remove this eslint comment and add onChange as dependency of the hook, or otherwise we could incur into unwanted behaviours. What do you think?

Reply:
At the current development stage of the Select component we can't add the onChange to the dependencies as it will create a memory leak inside the apps.
If you want I can add a comment over it to keep track of this issue

Reply:
You are right. A comment sounds good, so that we can easily remember why we didn't remove it and also that it needs a deeper study/refactor.

Reply:
Done